index.js

```diff
@@ -115,23 +117,81 @@ app.post('/customers/create', async (request, result) => {
   }
 })
 
-// Endpoint used to create a new inventory item.
-// Contains a site id (the site already exists in airtable).
+// Endpoint to create a new inventory item and an initial inventory update.
+// Assumes that the site, user, and product already exist in Airtable.
 app.post("/inventory/create", async (request, result) => {
   try {
-    const inventoryData = request.body;
-    // Remove the blank id field
-    delete inventoryData.id;
-    console.log("Inventory data: ", inventoryData);
+    const requestData = request.body;
+    const {userId, ...inventory} = requestData;
+    delete inventory.id; // Remove the blank id field
+    console.log("Inventory data: ", inventory);
+
+    const inventoryId = await createInventory(inventory);
+    console.log("Inventory id:", inventoryId);
+    console.log("Inventory created!\n\n");
+
+    const inventoryUpdate = {
+      userId: userId,
+      previousQuantity: 0,
+      updatedQuantity: inventory.currentQuantity,
+      inventoryId: inventoryId,
+      createdAt: moment().toISOString(),
+    }
+
+    console.log("Inventory Update data: ", inventory);
+    const inventoryUpdateId = await createInventoryUpdate(inventoryUpdate);
+    console.log("Inventory Update id:", inventoryUpdateId);
+    console.log("Inventory Update created!\n\n");
+
+    result.status(201);
+    result.json({ status: "OK", inventoryId, inventoryUpdateId });
+  } catch (err) {
+    console.log("Error when creating inventory and/or update: ", err);
+    result.status(400);
+    result.json({ error: err });
+  }
+});
+
+// Endpoint to create a new product, as well as a corresponding
+// inventory and inventory update record.
+// Assumes that the site and user exist in Airtable.
+app.post("/products/create", async (request, result) => {
```

Comment:
NIT: can you comment that the custom endpoint is needed to account for offline functionality (must be able to create all items in a single endpoint)?